src/analysis/processing/qgsalgorithmfilter.cpp

```diff
@@ -88,6 +88,7 @@ void QgsFilterAlgorithm::initAlgorithm( const QVariantMap &configuration )
 
 QVariantMap QgsFilterAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
 {
+  context.setInvalidGeometryCheck( QgsFeatureRequest::GeometryNoCheck );
```

Comment:
@nyalldawson is this the correct place to disable this check? I don't think it's used anywhere so far...